flaml/autogen/agent/user_proxy_agent.py

```diff
@@ -33,19 +36,61 @@ def __init__(
                     the number of auto reply reaches the max_consecutive_auto_reply.
                 (3) When "NEVER", the agent will never prompt for human input. Under this mode, the conversation stops
                     when the number of auto reply reaches the max_consecutive_auto_reply or when is_termination_msg is True.
+            function_map (dict[str, dict]): Mapping function names (passed to openai) to two types of functions:
+                    (1) A function to be called directly (dict): {
+                        "function" (Required, callable): a callable function that will be called
+                    }
+                    (2) A function in a class to be called (dict): {
+                        "class" (Required): an instance of a class.
+                        "func_name" (Optional, str): name of the function in the class. If not given the class will be called directly.
+                    }
+
+                Example 1:
+                def add_num_func(num_to_be_added):
+                    given_num = 10
+                    return num_to_be_added + given_num
+                oai_config = {"functions": [{"name": "add_num",...}]} # oai config, this will be passed to AssistantAgent
+                user = UserProxyAgent(name="test", function_map={"add_num": {"function": add_num_func}}) # pass in the function to the agent
+                func_call = {"name": "add_num", "args": {"num_to_be_added": 5}} # this is a potential function call passed from the LLM assistant
+                user._execute_function(func_call) # this will call add_num_func(5) and return 15
+
+                Example 2:
+                oai_config = {"functions": [{"name": "add_num",...}]}
+                # create the class instance and pass it to the agent
+                class AddNum:
+                    def __init__(self, given_num):
+                        self.given_num = given_num
+
+                    def add(self, num_to_be_added):
+                        self.given_num = num_to_be_added + self.given_num
+                        return self.given_num
+                user = UserProxyAgent(
+                    name="test",
+                    function_map={
+                        "add_num": {
+                            "class": AddNum(given_num=10),
```

Comment:
From this example, the value for "class" is actually an instance, not a class.
One thing I don't understand is why not also use `"function": AddNum(given_num=10).add`?

Reply:
You are right, there are no need for such a complex feature at all. I will simply this part.

Reply:
I have changed this back to previous implementation. User can just pass in `UserProxyAgent(function_map = {"func_name": callable_func})`, similarly `UserProxyAgent(function_map = {"func_name": class.callable_func})`.